virttest/qemu_devices/qbuses.py

```diff
@@ -632,9 +632,9 @@ def _set_device_props(self, device, addr):
         orig_addr = device.get_param('addr')
         if addr[1] or (isinstance(orig_addr, str) and
                        orig_addr.find('.') != -1):
-            device.set_param('addr', '%02x.%x' % (addr[0], addr[1]))
+            device.set_param('addr', '%s.%s' % (hex(addr[0]), hex(addr[1])))
```

Comment:
well having `addr=0x1d.0x7` looks really odd to me, but it is more accurate. Still I'd consider using `0x%x.x` which results in `addr=0x1d.7` which should be understood well. What do you think?